scripts/extract

```diff
@@ -31,7 +31,7 @@ def main(argv):
     with open(filename, 'rb') as fd:
```

Comment:
I'm not at all a Python expert, and even though I was able to make tests work (while providing a Python 3 approach) I'm not completely sure this is 100% OK.

Reply:
Is it not Python 3 -compatible as-is? I was sure I'd ran the scripts... might have skipped something.

Reply:
Unlike `print "x"` which is a syntax error in Python 3, `print >> sys.stderr` is valid syntax but will throw a type error when called (it's only called when the script is invoked incorrectly).